application/controllers/AdminController.php

```diff
@@ -498,6 +498,14 @@ public function getAdminFooter($url, $explanation, $return = false)
             $aData['versiontitle'] = gT('Version');
         }
 
+        $oEvent = new PluginEvent('beforeFooterRender');
```

Comment:
This is event is perhaps not needed if you use asset manager to register a script with current_plan instead?

https://github.com/LimeSurvey/cloud_client_limesurvey/pull/24/files#r1415473939

Reply:
@mohabmes Can you comment on why you marked this as resolved?

Reply:
Registering a script with the `current_plan` at this point serves no purpose; we only wanted that event to transmit the current plan for the partial rendering of the appropriate plan upgrade modal, displaying the accurate next bigger plan. (no JS involved).

https://github.com/LimeSurvey/cloud_client_limesurvey/blob/cf8134b0814d93a98659e8f5ad9c54364a070ac4/dbversion-600/application/views/admin/super/footer.php#L239

Reply:
Ah ok. In that case, it's probably better if the beforeFooterRender spits out HTML instead of a data array, so that other plugins can insert whatever HTML they want without changing any LS core view files.

That also means that the view file `planupgrade.php` can be moved to the LS Pro plugin folder instead of the core view folder.

Reply:
Compare with this event: https://manual.limesurvey.org/BeforeCloseHtml (only available in survey taking view, afaik; @Shnoulle you can confirm?).

So you can add a similar event to admin view called `beforeCloseAdminHtml`.

Reply:
Menu class : yes, it's interesting.

With public : we can rplace or add .twig view, but need netter too … with a way to really use twig block and extend

Reply:
Seems overkill for just adding some footer HTML, or?

Reply:
> Seems overkill for just adding some footer HTML, or?

I speak in general here :)